Angular: Ivy rendering for Canvas and Docs

ThibaudAV opened the pull request:
Issue: 🤔 

## What I did

new version of this [PR](https://github.com/storybookjs/storybook/pull/15002) but without new "docs to render stories out of context" changes 

Sorry about refactoring. 😈
But to many case to handle like navigate between canvas and docs view or some particularities on each one need to use some patten to try to have clean code and not many function intertwined.
I hope it's better

- For Canvas:
Generate an instance of render by id which allows to find it for a partial rendering. (when only the props change the application is not re-rendered)
Reset the PlatformBrowserDynamic before each rendering, if is already exist 

- For Docs:
Performs a full rendering with each change (Same behavior but without props optimization)
Reset the PlatformBrowserDynamic only when the user leaves the docs page
Must be called sequentially

works with ivy and this `storybook-addon-angular-ivy` addon too

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

ThibaudAV committed:
feat(angular): use new targetDOMNode to render docs

BREAKING CHANGE
@angular/elements and @webcomponents/custom-elements are now required as dependencies for angular app

shilman committed:
Add chromatic delay for angular ivy

shilman committed:
increase delay

ThibaudAV committed:
refactor: fix DeepScan

ThibaudAV committed:
test: fix angular test error

fix this error log
```
Failed to collect coverage from /tmp/storybook/app/angular/src/builders/build-storybook/index.spec.ts
ERROR: /tmp/storybook/app/angular/src/builders/build-storybook/index.spec.ts: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.
Invalid variable access: buildStandaloneMock
Allowed objects: Array, ArrayBuffer, Atomics, BigInt, BigInt64Array, BigUint64Array, Boolean, Buffer, DataView, Date, Error, EvalError, Float32Array, Float64Array, Function, GLOBAL, Generator, GeneratorFunction, Infinity, Int16Array, Int32Array, Int8Array, InternalError, Intl, JSON, Map, Math, NaN, Number, Object, Promise, Proxy, RangeError, ReferenceError, Reflect, RegExp, Set, SharedArrayBuffer, String, Symbol, SyntaxError, TextDecoder, TextEncoder, TypeError, URIError, URL, URLSearchParams, Uint16Array, Uint32Array, Uint8Array, Uint8ClampedArray, WeakMap, WeakSet, WebAssembly, arguments, clearImmediate, clearInterval, clearTimeout, console, decodeURI, decodeURIComponent, encodeURI, encodeURIComponent, escape, eval, expect, global, globalThis, isFinite, isNaN, jest, parseFloat, parseInt, process, queueMicrotask, require, root, setImmediate, setInterval, setTimeout, undefined, unescape.
Note: This is a precaution to guard against uninitialized mock variables. If it is ensured that the mock is required lazily, variable names prefixed with `mock` (case insensitive) are permitted.

   6 | const buildStandaloneMock = jest.fn().mockImplementation((_options: unknown) => Promise.resolve());
   7 |
>  8 | jest.mock('@storybook/angular/standalone', () => buildStandaloneMock);
```

tmeasday commented on app/angular/src/client/preview/render.ts:
```diff
@@ -1,32 +1,41 @@
+  targetDOMNode: HTMLElement;
+  id: string;
 }) {
-  showMain();
```
This is what is tripping up Chromatic -- we need to call `showMain()` at some point to remove the `.sb-show-nopreview` class from the body.

shilman replied:
Nice catch!

ThibaudAV replied:
Nice, I must have lost it I don't know how

ThibaudAV committed:
fix: call showMain in main render function

We need to call showMain() at some point to remove the .sb-show-nopreview class from the body.

ThibaudAV committed:
Merge branch 'next' into angular/renderer-v2

# Conflicts:
#	app/angular/package.json
#	yarn.lock

ThibaudAV committed:
Revert "increase delay"

This reverts commit 83e02f0df743443960688cab0b3af3db82cf964c.

ThibaudAV committed:
Revert "Add chromatic delay for angular ivy"

This reverts commit 097961c10e903faba892defcea29b4c57ddcb865.

shilman reviewed (approved):
I'm in love, again. 😻
